torchvision/datasets/video_utils.py

```diff
@@ -366,6 +371,11 @@ def get_clip(self, idx: int) -> Tuple[torch.Tensor, torch.Tensor, Dict[str, Any]
             video = video[resampling_idx]
             info["video_fps"] = self.frame_rate
         assert len(video) == self.num_frames, f"{video.shape} x {self.num_frames}"
+
+        if self.output_format == "TCHW":
+            # [T,H,W,C] --> [T,C,H,W]
+            video = video.permute(0, 3, 1, 2)
```

Comment:
I think this might not work when `backend` != pyav. The `_read_video_from_file()` returns `Tensor[T, H, W, C]`, so we need to enforce the right format depending on the backend. This requires a bit more logic.

Reply:
I think the easiest way to do this is pass the parameter to `read_video()` at line 332 and potentially add a similar param to `_read_video_from_file()`.

Reply:
Are you sure? I'm converting the format at the very end, after both backend have done their magic.

> The _read_video_from_file() returns Tensor[T, H, W, C]

Agreed and this is what `read_video()` with `pyav` should return as well.

Reply:
The 2 backends being used is precisely why I didn't pass `read_video(output_format=output_format)` BTW

Reply:
You are right. Somehow I got confused and thought the two backends returned different order of dims. What you have will work. I'll check again on Monday with clearer head and merge.

Out of curiosity, what's the reason you opted for adding the parameter to `read_video()` but not to `_read_video_from_file()` and handle this in there uniformly?

Reply:
> adding the parameter to read_video() but not to _read_video_from_file() and handle this in there uniformly

I added this to all public-facing video readers (I think?) but I didn't want to bother with the private ones.